.travis.yml

```diff
@@ -1,91 +1,59 @@
-language: python
+language: minimal
+dist: xenial
+services: docker
 
-install:
-    - sudo apt-get -y -q update
-    - sudo apt-get -y -q build-dep herbstluftwm
-    - sudo apt-get -y -q install cmake ninja-build dzen2 xdotool
+
+before_install:
+    # Delete .git/ to maximize ccache effectiveness (no version string mismatches)
+    - rm -rf .git
 
 jobs:
     include:
-        - name: "Ubuntu 16.04, gcc-8, Debug build, run tests"
-          dist: xenial
-          python: "3.6"
-          before_install:
-              - ./ci-check-using-std.sh
-              - export CC=gcc-8 CXX=g++-8
-              - linkerfix="-fuse-ld=gold" # needed to avoid linker error
-              - export LDFLAGS="$linkerfix" LDXXFLAGS="$linkerfix"
-          addons:
-              apt:
-                  sources:
-                      - ubuntu-toolchain-r-test
-                  packages:
-                      - g++-8
-                      - xterm
-                      - tox
+        - name: "Build with most recent GCC & Clang, run checks and tests"
+          env: TRAVIS_CACHE_ID=disco
+          cache:
+              directories:
+                  - $HOME/.ccache-clang-7
+                  - $HOME/.ccache-gcc-8
+                  - $HOME/.tox-cache
           script:
-              - cmake -G'Ninja' -DCMAKE_BUILD_TYPE=Debug . # run tests with sanitizer
-              - ninja -v -k 10
-              - tox -- -n auto -v
+              # Prepare Docker image with bleeding edge Ubuntu 19.04 (disco)
+              - docker pull dnnr/hlwm-ci:disco
```

Comment:
Note that this is pulling a reference image from my personal Docker Hub repository.
We'll probably want to switch to a generic org if we're going to merge this (I'm already squatting `hlwm` for this).

Reply:
Btw, because it's not really obvious: The images at `dnnr/hlwm-ci` get built automatically by Docker, through a connection with my GitHub repo. So while this *is* an external artifact, it does not need to be maintained manually.